content/how-to/gql/call-the-graphql-api.md

```diff
@@ -10,40 +10,434 @@ menu:
+
+
+For example, this query places the ID of the resource that it requests as an inline variable:
+
+```graphql
+query myCustomName {
+  getEquipment(id: "Kitchen_mixer_b_01") {
+    _createdBy
+  }
+```
+
+Instead, you can pass this argument as a variable.
+This requires the following changes:
+
+1. In the argument for your query, name the variable and state its type.
+This instructs the query to receive data from outside of its context:
+
+    ```graphql
+    ## Name variable and type
+    query myCustomName ($getEquipmentId: String) {
+      ## operations go here.
+    }
+    ```
+1. In the operation, pass the variable as a value in the argument.
+So, to add the variable as a value to the `id` key:
+
+    ```graphql
+    query GetEquipment($getEquipmentId: String) {
+    
+      ## pass variable to one or more operations
+      getEquipment(id: $getEquipmentId) {
+        ## fields go here
+    }
+    ```
+    
+1. In a separate, `variables` section of the query define the JSON object that is your variable:
+
+   ```json
+   {
+    "getEquipmentId": "Kitchen_mixer_b_01"
+   }
+   ```
+
+
+{{< tabs >}}
+{{% tab "Query" %}}
+    
+```graphql
+query GetEquipment($getEquipmentId: String) {
+  getEquipment(id: $getEquipmentId) {
+    _createdBy
+  }
+}
+```
+{{% /tab %}}
+{{% tab "variables" %}}
+
+```json
+{
+ "getEquipmentId": "Kitchen_mixer_b_01"
+}
+```
+{{% /tab %}}
+{{% /tab %}}
+
+The preceding example is minimal, but the use of variables to _parameterize_ also applies to complex object creation and filtering.
+For example, this _mutation_ uses variables to create an array of Persons:
+
+{{< tabs >}}
+  {{% tab "Mutation" %}}
+```graphql
+mutation AddPerson($input: [AddPersonInput!]!) {
+  addPerson(input: $input) {
+    person {
+      id
+    }
+  }
+}
+```
+{{% /tab %}}
+{{% tab "variables" %}}
+```json
+{
+  "input": [
+ {"id": "234", "label":"Jan Smith"},
+ {"id": "235", "label": "John Ramirez"}
+  ]
+}
+```
+{{% /tab %}}
+{{< /tabs >}}
+
+To learn more, read the offical GraphQL documentation on [Variables](https://graphql.org/learn/queries/#variables).
+
+## Shortcuts to make calls more manageable
+
+The following sections provide some common ways to reduce boilerplate and shorten the necessary coding for a call.
+
+### Template requested fields with fragments
+
+Along with [variables](#variables), you can use _Fragments_ to reduce repetitive writing.
+
+Fragments are common fields that you use when querying an object.
+For example, imagine you wanted to make queries to different equipment objects for their `id`, `label`, `_createdBy`, and `versions[]` properties.
+Instead of writing these fields in each operation, you could define them in a fragment then refer to that fragment in each specific operation or query.
```

Comment:
```suggestion
Instead of writing these fields in each operation, you could define them in a fragment, and then refer to that fragment in each specific operation or query.
```